core/domain/activity_jobs_one_off.py

```diff
@@ -628,3 +629,110 @@ def reduce(key, values):
+            'id_string_format': 'question-%s-%s',
+            'id_field': 'question_id'
+        },
+        'SkillSnapshotMetadataModel': {
+            'parent_model_class': skill_models.SkillModel,
+            'commit_log_model_class': skill_models.SkillCommitLogEntryModel,
+            'id_string_format': 'skill-%s-%s',
+            'id_field': 'skill_id'
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s',
+            'id_field': 'exploration_id'
+
+        }
+    }
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        job_class = AddMissingCommitLogsJob
+        return job_class.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS
+
+    @staticmethod
+    def map(snapshot_model):
+        job_class = AddMissingCommitLogsJob
+        class_name = snapshot_model.__class__.__name__
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_properties = job_class.MODEL_NAMES_TO_PROPERTIES[class_name]
+        commit_log_id = (
+            model_properties['id_string_format'] % (model_id, version))
+        commit_log_model = (
+            model_properties['commit_log_model_class'].get_by_id(
+                commit_log_id))
+        commit_logs_should_exist = True
+
+        parent_model = (
+            model_properties['parent_model_class'].get_by_id(model_id))
+        if class_name == 'ExplorationRightsSnapshotMetadataModel':
+            if snapshot_model.commit_type in ['create', 'delete']:
+                commit_logs_should_exist = False
+
+        if commit_log_model is None and commit_logs_should_exist:
+            if parent_model is not None:
+                # Public is the default value for post_commit_status for the
+                # question and skill models. For the exp model, it is assigned
+                # from the rights model.
+                commit_log_model = model_properties['commit_log_model_class'](
+                    id=commit_log_id,
+                    user_id=snapshot_model.committer_id,
+                    commit_type=snapshot_model.commit_type,
+                    commit_message=snapshot_model.commit_message,
+                    commit_cmds=snapshot_model.commit_cmds,
+                    post_commit_status=constants.ACTIVITY_STATUS_PUBLIC,
+                    version=int(version)
+                )
+                setattr(
+                    commit_log_model, model_properties['id_field'], model_id)
+                if class_name == 'ExplorationRightsSnapshotMetadataModel':
+                    commit_log_model.post_commit_status = parent_model.status
+                elif class_name == 'ExplorationSnapshotMetadataModel':
+                    rights_model = exp_models.ExplorationRightsModel.get_by_id(
```

Comment:
This is the current version of the rights model, but the commit that you are modifying may be a lot older, when the rights model was different, correct? So you should probably be retrieving the version of the rights model that is appropriate to that phase instead...?

Same concern for "parent model" above.

Reply:
Done.